### ISSUE TITLE
Fish: `fnm use` doesn't respect $FNM_VERSION_FILE_STRATEGY when terminal session starts from inside a directory with `.nvmrc`

### ISSUE THREAD

**Issue by SirWrexes** [Schniz/fnm#964]
Greetings!
As title says, despite having set `FNM_VERSION_FILE_STRATEGY=recursive`, `fnm use` and the `--use-on-cd` installation doesn't always take the environment variable in account.

In my `~/.config/fish/config.fish` I have the following:
```fish
command -q fnm
    and fnm env --use-on-cd | source
    and set -e FNM_VERSION_FILE_STRATEGY
    and set -Ux FNM_VERSION_FILE_STRATEGY recursive
```
When checking the variable's value, I get the correct setting (ignore the silly greeting :smile:):
![image](https://github.com/Schniz/fnm/assets/5567164/fc14044f-b7dd-4ee2-9d71-62602f260c70)

However neither `fnm use` nor changing directories looks for `.nvmrc` in a parent directory *if my terminal session starts from inside a one using `.nvmrc` or one of its subdirectories*

Here I create the dirs and file:
![image](https://github.com/Schniz/fnm/assets/5567164/7228949a-e70a-4d19-89d1-9e9cc8413b8c)

Then if I go in there and move around it works just fine:
![image](https://github.com/Schniz/fnm/assets/5567164/5c4de04e-2ada-4605-b6d1-9844a4394e88)

However, if I start a new instance of my terminal from inside, it becomes somewhat clunky:
![image](https://github.com/Schniz/fnm/assets/5567164/fcbc2ee4-0012-4a34-8a37-44fbc390ed72)

Coming out the `root` directory and back in doesn't fix the issue either.
![image](https://github.com/Schniz/fnm/assets/5567164/90e2fa05-1a52-4637-93a3-eccf382818f0)

Appending `--version-file-strategy $FNM_VERSION_FILE_STRATEGY` explicity to `fnm use` inside `_fnm_autoload_hook` seems to fix the issue.
Replacing my setup in `~/.config/fish/config.fish` for demonstration purposes, then opening a new terminal tab inside `root`:
```fish
command -q fnm
#     and fnm env --use-on-cd | source
    and echo 'set -gx PATH "/run/user/1000/fnm_multishells/60026_1685202649680/bin" $PATH;
    set -gx FNM_ARCH "x64";
    set -gx FNM_MULTISHELL_PATH "/run/user/1000/fnm_multishells/60026_1685202649680";
    set -gx FNM_VERSION_FILE_STRATEGY "recursive";
    set -gx FNM_LOGLEVEL "info";
    set -gx FNM_DIR "/home/wrexes/.local/share/fnm";
    set -gx FNM_NODE_DIST_MIRROR "https://nodejs.org/dist";
    function _fnm_autoload_hook --on-variable PWD --description \'Change Node version on directory change\'
        status --is-command-substitution; and return
        fnm use --silent-if-unchanged --version-file-strategy $FNM_VERSION_FILE_STRATEGY
    end

    _fnm_autoload_hook
    ' | source
    and set -e FNM_VERSION_FILE_STRATEGY
    and set -Ux FNM_VERSION_FILE_STRATEGY recursive
```
![image](https://github.com/Schniz/fnm/assets/5567164/82e0d6bb-ef96-4261-9fbd-144a5acba871)

**Comment by Schniz** [comment]
Is there a reason why not to do `fnm env —version-file-strategy=recursive` and setting it up through the env vars? This would be cleaner and will fix your issue

**Comment by Schniz** [comment]
The `eval` (or `source` in fish) runs the `fnm use`, which happens before you set the env var. it’s not a bug

**Comment by SirWrexes** [comment]
Ah.
So that means I just need to set the variable beforehand and I'm a doofus ?
Somehow when I tested in my CLI it seemed like it wouldn't work, but with the config file everything seems fine...
Welp, sorry for wasting your time, thanks for answering!

**Comment by Schniz** [comment]
the best way would be to use `fnm env --version-file-strategy=recursive` as it will output the required env var for you. the env vars are implementation details to store the config/state in some way in the current shell <picture data-single-emoji=":relieved_galactic:" title=":relieved_galactic:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/galactic_relifed/33851380ad8f06bb.png" alt=":relieved_galactic:" align="absmiddle"></picture>

**Comment by Schniz** [comment]
```sh-session
$ fnm env | rg STRATEGY
export FNM_VERSION_FILE_STRATEGY="recursive"
```

**Comment by SirWrexes** [comment]
Yeah I simply missed that option when setting things up. I was using `fnm` before the version file strategy was implemented, and had to set it up in a rush as my project mates don't use Fish so they have `nvm` and this is its default behaviour.

I saw no mention of this option for `fnm env` in the README, and didn't check out the extended help for commands. Mistake on my part. :+1: 

By the way, since it replicates `nvm`'s default behaviour, some people are going to want it for similar reasons and might do the same mistake as me. Shouldn't it be visible from the repo home ?

**Comment by Schniz** [comment]
I am thinking of making it the default, because it makes more sense and it is not slow.
the less config, the better

**Comment by SirWrexes** [comment]
Sounds like a good plan, IMHO.
I guess we're done here, have a good day!

**Comment by mugilan-viewzen** [comment]
Thank you, this solved my issue

`eval "$(fnm env --use-on-cd --version-file-strategy=recursive)"`
